conda/models/environment.py

```diff
@@ -0,0 +1,151 @@
+    external_packages: dict[str, list[Any]] = field(default_factory=dict)
+
+    #: The complete list of specs for the environment.
+    #: eg. after a solve, or from an explicit environemnt spec
+    explicit_packages: list[PackageRecord] = field(default_factory=list)
+
+    #: Environment name
+    name: str | None = None
+
+    #: User requested specs for this environment.
+    requested_packages: list[MatchSpec] = field(default_factory=list)
+
+    #: Environment variables to be applied to the environment.
+    variables: dict[str, str] = field(default_factory=dict)
+
+    def __post_init__(self):
+        # an environment must have a name of prefix
+        if not self.prefix:
+            raise CondaValueError("'Environment' needs a 'prefix'.")
+
+    @classmethod
+    def merge(cls, *environments):
+        """
+        **Beta** While in beta, expect both major and minor changes across minor releases.
+
+        Merges multiple environments into a single environment following the rules:
+        * Keeps first name and/or prefix.
+        * Concatenates and deduplicates requirements.
+        * Reduces configuration and variables (last key wins).
+        """
+        name = None
+        prefix = None
+        platform = None
+        names = [env.name for env in environments if env.name]
+        prefixes = [env.prefix for env in environments if env.prefix]
+
+        if names:
+            name = names[0]
+            if len(names) > 1:
+                log.debug("Several names passed %s. Picking first one %s", names, name)
+
+        if prefixes:
+            prefix = prefixes[0]
+            if len(prefixes) > 1:
+                log.debug(
+                    "Several prefixes passed %s. Picking first one %s", prefixes, prefix
+                )
+
+        platforms = [env.platform for env in environments if env.platform]
+        # Ensure that all environments have the same platform
+        if len(set(platforms)) == 1:
+            platform = platforms[0]
+        else:
+            raise CondaValueError(
+                f"Conda can not merge environments of different platforms. Recieved environments with plafroms {platforms}"
+            )
+
+        requested_packages = list(
+            dict.fromkeys(
+                requirement
+                for env in environments
+                for requirement in env.requested_packages
+            )
+        )
+
+        explicit_packages = list(
+            dict.fromkeys(
+                requirement
+                for env in environments
+                for requirement in env.explicit_packages
+            )
+        )
+
+        variables = {k: v for env in environments for (k, v) in env.variables.items()}
+
+        config = {}
+        external_packages = {}
+        for env in environments:
+            # Config items can be any type, merge them so that lists get
+            # concatenated, dicts get merged, and primitive types get clobbered.
```

Comment:
Just highlighting that the `.merge()` logic in [`conda.common.configuration`](https://github.com/conda/conda/blob/main/conda/common/configuration.py) seems to be a bit more involved than this. Maybe this simplification is ok for the involved keys, but in that case we should make sure we are erroring out on unexpected keys.

Reply:
The configuration class adds additional complexity in merging by adding logic for things like `final` markers in `.condarc` configs.  This kind of information will be lost is the context is serialized to a simple dict.